api-reference/50 Common/utils/zz Errors and Warnings/W0017.md

```diff
@@ -3,7 +3,30 @@ id: ErrorsCore.W0017
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Occurs a component's **width** or **height** properties have a value of the `function` type.
 
 ---
-<!-- Description goes here -->
\ No newline at end of file
+If you need to specify a component's height or width and make the component's size responsive, use viewport units instead of a function. The most simple replacement is the following:
+
+    <!--JavaScript-->
+    // A value of the deprecated function type 
+    width: function() {
+        return window.innerWidth / 1.5;
+    }
+
+    // Specify the same width in viewport units as a string
+    width: '66vw'
+
+If you used a function for more complex calculations, replace it with the native CSS <a href="https://developer.mozilla.org/en-US/docs/Web/CSS/calc()" target="_blank">calc()</a> function as follows:
+
+    <!--JavaScript-->
+    // A value of the deprecated function type
+    height: function() {
+        return window.innerHeight / 2 + 100;
+    }
+
```

Comment:
This should prevent the JavaScript and CSS from being collected into clickable tabs:

```suggestion

<!-- -->

```